DNN Platform/Library/Entities/Portals/PortalSettingsManager.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+namespace DotNetNuke.Entities.Portals
+{
+    using System;
+    using System.Collections.Generic;
+
+    using DotNetNuke.Abstractions.Portals;
+    using DotNetNuke.Abstractions.Settings;
+    using DotNetNuke.Common.Utilities;
+    using DotNetNuke.Data;
+    using DotNetNuke.Services.Exceptions;
+    using DotNetNuke.Services.Log.EventLog;
+
+    /// <inheritdoc />
+    public class PortalSettingsManager : IPortalSettingsManager
+    {
+        /// <inheritdoc />
+        public ISettingsService GetPortalSettings(int portalId)
+        {
+            return this.GetPortalSettings(portalId, string.Empty);
+        }
+
+        /// <inheritdoc />
+        public ISettingsService GetPortalSettings(int portalId, string cultureCode)
+        {
+            if (string.IsNullOrEmpty(cultureCode) && portalId > -1)
+            {
+                cultureCode = PortalController.GetActivePortalLanguage(portalId);
+            }
+
+            var cachedSettings = this.GetCachedSettings(portalId, cultureCode);
+            var saveService = new PortalSaveSettingsService(portalId, cultureCode);
+            var deleteService = new PortalDeleteSettingsService(portalId, cultureCode);
+
+            return new PortalSettingsService(() => this.GetCachedSettings(portalId, cultureCode), saveService, deleteService);
+        }
```

Comment:
This may be one of the focal points of this code review and I would like to highlight the change as well as my reasoning behind the change. 

I wanted to implement the `PortalSettingsService` as a very simple wrapper around an `IDictionary<string, string>`. Since DNN has static API calls everywhere I didn't want the `PortalSettingsService` to be tightly coupled to anything. In theory this implementation could live in a .NET Standard 2.0 project and doesn't need to exist in the `DotNetNuke.Library` project. To do this I needed to create an `ISaveSettingsService` and a `IDeleteSettingsService` The implementations of this interfaces are passed in at instantiation time. This allows the wrapper (`PortalSettingsService`) to easily invoke the functionality. 

The other item here is the callback function that retrieves the latest settings `GetCacheSettings(int, int)`. This is useful in the `PortalSettingsService` to reset the local `IDictionary<string, string>` whenever the cache is cleared. Due to the de-coupled structure we need a way for it to easily reload the data. Since the `PortalSettingsService` doesn't have direct knowledge how to construct the `IDictionary<string, string>` we are using the callback that stores the reference on how to load the data.